From c56194d122b0308c7c3b827a42dfdb700aa05a76 Mon Sep 17 00:00:00 2001
From: xxm <xxm@rock-chips.com>
Date: Thu, 8 May 2014 10:26:19 +0800
Subject: [PATCH] rockchip:midgard:mutex unlock pm.lock after
 kbase_pm_do_poweron when kbase_pm_resume to avoid boring log

---
 drivers/gpu/arm/midgard/mali_kbase_pm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/arm/midgard/mali_kbase_pm.c b/drivers/gpu/arm/midgard/mali_kbase_pm.c
index 212221f94fda..a0cd00b251b4 100755
--- a/drivers/gpu/arm/midgard/mali_kbase_pm.c
+++ b/drivers/gpu/arm/midgard/mali_kbase_pm.c
@@ -427,10 +427,11 @@ void kbase_pm_resume(struct kbase_device *kbdev)
 	/* MUST happen before any pm_context_active calls occur */
 	mutex_lock(&kbdev->pm.lock);
 	kbdev->pm.suspending = MALI_FALSE;
-	mutex_unlock(&kbdev->pm.lock);
 
 	kbase_pm_do_poweron(kbdev, MALI_TRUE);
 
+	mutex_unlock(&kbdev->pm.lock);
+	
 	/* Restart PM Metric timer on resume */
 	kbasep_pm_metrics_init(kbdev);
 	kbasep_pm_record_gpu_idle(kbdev);
-- 
2.35.3

